src/txmempool.cpp

```diff
@@ -1244,6 +1246,64 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
     return total;
 }
 
+void CTxMemPool::UpdateMetrics() const {
+    LOCK(cs);
+
+    // For transactions that pay less than the conventional fee, track the sum
+    // of unpaid actions (as they are subject to the unpaid action limit).
+    // Lower boundaries are inclusive (i.e. a transaction with weight 0.25 is
+    // included in `weightRatio25to49`).
+    size_t weightRatio00to24 = 0;
+    size_t weightRatio25to49 = 0;
+    size_t weightRatio50to74 = 0;
+    size_t weightRatio75to99 = 0;
+
+    // For transactions that pay at least the conventional fee, track the sum of
+    // transaction sizes (the metric by which they are limited) across several
+    // buckets. Upper boundaries are inclusive (i.e. a transaction with weight 2
+    // is included in `weightRatio1to2`).
+    size_t weightRatio1to2 = 0;
+    size_t weightRatio2to3 = 0;
+    size_t weightRatio3to4 = 0;
+
+    size_t totalActions = 0;
+    for (auto entry : mapTx) {
+        totalActions += std::max(GRACE_ACTIONS, entry.GetTx().GetLogicalActionCount());
+        int128_t weightRatio = entry.GetWeightRatio();
+        if (weightRatio >= WEIGHT_RATIO_SCALE) {
+            if (weightRatio > 3 * WEIGHT_RATIO_SCALE) {
+                weightRatio3to4 += entry.GetTxSize();
+            } else if (weightRatio > 2 * WEIGHT_RATIO_SCALE) {
+                weightRatio2to3 += entry.GetTxSize();
+            } else {
+                weightRatio1to2 += entry.GetTxSize();
+            }
+        } else {
+            if (weightRatio >= 3 * WEIGHT_RATIO_SCALE / 4) {
+                weightRatio75to99 += entry.GetUnpaidActionCount();
+            } else if (weightRatio >= WEIGHT_RATIO_SCALE / 2) {
+                weightRatio50to74 += entry.GetUnpaidActionCount();
+            } else if (weightRatio >= WEIGHT_RATIO_SCALE / 4) {
+                weightRatio25to49 += entry.GetUnpaidActionCount();
+            } else {
+                weightRatio00to24 += entry.GetUnpaidActionCount();
+            }
+        }
+    }
+
+    MetricsGauge("zcash.mempool.actions.unpaid", weightRatio00to24, "lt", "0.25");
+    MetricsGauge("zcash.mempool.actions.unpaid", weightRatio25to49, "lt", "0.5");
+    MetricsGauge("zcash.mempool.actions.unpaid", weightRatio50to74, "lt", "0.75");
+    MetricsGauge("zcash.mempool.actions.unpaid", weightRatio75to99, "lt", "1");
+    MetricsGauge("zcash.mempool.actions.total", totalActions);
+    MetricsGauge("zcash.mempool.size.transactions", size());
+    MetricsGauge("zcash.mempool.size.weighted", weightRatio1to2, "le", "2");
+    MetricsGauge("zcash.mempool.size.weighted", weightRatio2to3, "le", "3");
+    MetricsGauge("zcash.mempool.size.weighted", weightRatio3to4, "le", "4");
+    MetricsGauge("zcash.mempool.size.bytes", GetTotalTxSize());
```

Comment:
Similarly here, we currently give the total tx size as a metric, so we can derive the sum of "unpaid" transaction sizes via subtraction, but maybe we instead want it to be an explicit metric? That would be overall duplicative, but `zcash.mempool.size.bytes` is an existing metric that we should retain.